crest/Assets/Crest/Crest/Shaders/OceanInputs/GerstnerShared.hlsl

```diff
@@ -91,3 +88,24 @@ half4 ComputeGerstner(float2 worldPosXZ, float3 uv_slice)
 
 	return _Weight * half4(result, sss);
 }
+
+half4 ComputeShorelineGerstner(float2 worldPosXZ, float3 uv_slice, half depth, half distanceToShore, half2 directionToShore)
+{
+	float2 displacementNormalized = 0.0;
+	half3 result = (half3)0.0;
+
+	if(depth > 0.0)
+	{
+		const float pi = 3.141;
+		const float twoPiOverWavelength = (2.0 * pi) / 2.5;
+		const float twoPiOverPeriod = (2.0 * pi) / 2.0;
+		const float amplitude = 0.12;
+		// Chop increases as depth increases
+		const float chopAmplitude = 2.0/(1.0+sqrt(distanceToShore));
```

Comment:
You will notice sqrt(distanceToShore) is used as a paramater to increase the chop as the distance to the shore increases - another option here could be to use depth - but this worked well in the experimentation.